### PR TITLE
[Mellanox] Add hw-mgmt patch for SimX platform adaptation

### PR DESCRIPTION
**Pull request by shlomibitton**
Signed-off-by: Shlomi Bitton <shlomibi@nvidia.com>

#### Why I did it
System is stuck on 'starting' state on SimX platform because of infinite loop on 'hw-management-ready.sh' script .
The loop is polling to check if the hw-mgmt sysfs created before proceeding with the flow, for SimX platform the sysfs will never create so the system is not starting properly.

#### How I did it
Add a condition to poll on hw-mgmt sysfs only if the switch is real HW and not SimX platform.

#### How to verify it
Check "systemctl status hw-management.service" output on a SimX switch with this patch, the state will be "active".

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [x] 202012

### CODE REVIEW

**Review by liat-grozovik — DISMISSED** [review]

File: platform/mellanox/hw-management/0003-Make-hw-mgmt-SimX-compatiable.patch [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ fi
+ 
+-while [ ! -d /sys/devices/platform/mlxplat/mlxreg-hotplug/hwmon ]
+-do
+-	sleep 1
+-done
++if [ "$(cat /sys/devices/virtual/dmi/id/sys_vendor)" != "QEMU" ]; then
++	while [ ! -d /sys/devices/platform/mlxplat/mlxreg-hotplug/hwmon ]
++	do
++		sleep 1
```
Comment (nazariig) [inline platform/mellanox/hw-management/0003-Make-hw-mgmt-SimX-compatiable.patch:16]: @shlomibitton why do we need to sleep? Why not simply return? Does it mean that hw-management service will always be in starting condition?

Comment (shlomibitton) [reply platform/mellanox/hw-management/0003-Make-hw-mgmt-SimX-compatiable.patch:16]: The loop is polling to check if the hw-mgmt sysfs created before proceeding with the flow.
AFAIK, we want to wait until it is valid before exiting this script, this is the previous implementation.
The fix is only to skip it if this is a SimX platform.

**Comment by dprital** [comment]
@lguohan - Can you please review ?

File: platform/mellanox/hw-management/0003-Make-hw-mgmt-SimX-compatiable.patch (hunk 1 above)
Comment (nazariig) [reply platform/mellanox/hw-management/0003-Make-hw-mgmt-SimX-compatiable.patch:16]: @shlomibitton `"$(cat /sys/devices/virtual/dmi/id/sys_vendor)" != "QEMU"` this doesn't work anymore. Please use `lspci` instead.

**Review by nazariig — CHANGES_REQUESTED** [review]

File: platform/mellanox/hw-management/0003-Make-hw-mgmt-SimX-compatiable.patch [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+index 70f1297..bb45ab9 100755
+--- a/usr/usr/bin/hw-management.sh
++++ b/usr/usr/bin/hw-management.sh
+@@ -1110,6 +1110,13 @@ do_chip_down()
+ 	/usr/bin/hw-management-thermal-events.sh change hotplug_asic down %S %p
+ }
+ 
++check_simx()
++{
++        if lspci -vvv | grep SimX &> /dev/null; then
```
Comment (nazariig) [inline platform/mellanox/hw-management/0003-Make-hw-mgmt-SimX-compatiable.patch:32]: @shlomibitton `if [ -n "$(lspci -vvv | grep SimX)" ]; then`

File: platform/mellanox/hw-management/0003-Make-hw-mgmt-SimX-compatiable.patch [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
++++ b/usr/usr/bin/hw-management-ready.sh
+@@ -49,9 +49,12 @@ if [ -d /var/run/hw-management ]; then
+ 	rm -fr /var/run/hw-management
+ fi
+ 
+-while [ ! -d /sys/devices/platform/mlxplat/mlxreg-hotplug/hwmon ]
+-do
+-	sleep 1
+-done
++if [ "$(lspci -vvv | grep SimX)" == "" ]; then
```
Comment (nazariig) [inline platform/mellanox/hw-management/0003-Make-hw-mgmt-SimX-compatiable.patch:13]: @shlomibitton `if [ -z "$(lspci -vvv | grep SimX)" ]; then`

**Review by nazariig — APPROVED** [review]

**Review by liat-grozovik — APPROVED** [review]

**Pull request merged**